x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/inference/TextEmbedding.java

```diff
@@ -40,23 +40,32 @@ public class TextEmbedding extends InferenceFunction<TextEmbedding> {
 
     @FunctionInfo(
         returnType = "dense_vector",
-        description = "Generates dense vector embeddings for text using a specified inference endpoint.",
-        appliesTo = { @FunctionAppliesTo(lifeCycle = FunctionAppliesToLifecycle.DEVELOPMENT) },
+        description = "Converts a constant text into dense vector embeddings using a inference endpoint. "
+            + "The resulting vectors can be used for knn search, and other vector-based operations. "
+            + "Requires an inference endpoint configured with the `text_embedding` task type. "
+            + "See [Inference API documentation](docs-content://explore-analyze/elastic-inference/inference-api.md) "
+            + "for how to create inference endpoints.",
+        appliesTo = { @FunctionAppliesTo(version = "9.3", lifeCycle = FunctionAppliesToLifecycle.PREVIEW) },
         preview = true,
         examples = {
             @Example(
                 description = "Generate text embeddings using the 'test_dense_inference' inference endpoint.",
                 file = "text-embedding",
-                tag = "embedding-eval"
+                tag = "text-embedding-eval"
             ) }
     )
     public TextEmbedding(
         Source source,
-        @Param(name = "text", type = { "keyword" }, description = "Text to generate embeddings from") Expression inputText,
+        @Param(
+            name = "text",
+            type = { "keyword" },
+            description = "Text to generate embeddings from. Must be a non-null keyword constant."
```

Comment:
```suggestion
            description = "Text to generate embeddings from. Must be a non-null literal string value."
```

to my eyes that's clearer, but maybe it's not technically precise, I'll let you decide :)